client/ayon_core/hosts/houdini/api/plugin.py

```diff
@@ -237,14 +238,18 @@ def collect_instances(self):
 
             node_data = read(instance)
 
+            for key, value in tuple(node_data.items()):
+                if not key.startswith(KEY_PREFIX):
+                    continue
+                new_key = key[len(KEY_PREFIX):]
+                node_data[new_key] = node_data.pop(key)
```

Comment:
So to be clear - just for backwards compatibility we're still reading the full `node_data` but just allow `AYON_` prefixed keys to override the collected data`.

This does help with backwards compatibility to load them - but will from that moment after the first 'update' have duplicated data where the old will remain lingering around. Yes?

### Setting a decent label?

Another note is that with this change now this in the Houdini UI will now all get `AYON_` prefixes on the labels, making them quite long and not as nice visually:

![image](https://github.com/ynput/ayon-core/assets/2439881/dfeecf1c-b994-4d02-aad5-a4a8dac791ce)

I'm wondering, if instead we should maybe update the `imprint` function to support setting labels for the attributes where it e.g. strips of `AYON_` for the nice human readable label but keeps the `AYON_` prefix for the attribute name itself.


```python
# pseudocode
def imprint(node, data, labeler=None):
    for attr_name, attr_value in data.items():
        attr_label = None
        if labeler is not None:
            attr_label = labeler(attr_name)
        set_attribute(node, attr_name, value, label=attr_label)


def labeler(name):
    """Label 'AYON_creator_attributes' as 'Creator Attributes'"""

    def uppercase_first(word):
        if not word:
            return word    
        return word[0].upper() + word[1:]

    name = name.removeprefix(KEY_PREFIX)
    
    return " ".join(uppercase_first(part) for part in split("_"))


# For example
imprint(node, {
    "task": "modeling",
    "creator_attributes": "hello",
    "publish_attributes": "boo"
}, labeler=labeler)
```

Reply:
> I'm wondering, if instead we should maybe update the imprint function to support setting labels for the attributes where it e.g. strips of AYON_ for the nice human readable label but keeps the AYON_ prefix for the attribute name itself.

I tried to implement it here https://github.com/ynput/ayon-core/pull/124

Reply:
@MustafaJafar can you try and change the target branch to this one - just so it shows just the diff to this one.

Reply:
I will but, I couldn't see any difference 🤔

Reply:
That's odd - does it contain that much more changes than this PR?

Ah wait - it does. Why does it?